Publish/Subscribe: runtime error on channel close

Consider the following snippet

```
func Subscribe(channel string, subscr Subscriber) (quit chan int) {
    sub := make(chan string, 1)
    quit = make(chan int)
    messages := make(chan redis.Message, 0)

    go func() {
        if err := client.Subscribe(sub, nil, nil, nil, messages); err != nil {
            log.Error(err)
        }
        fmt.Println("UNBLOCKED")
    }()

    sub <- channel

    go func() {
    L:
        for {
            select {
            case <-quit:
                break L
            case msg := <-messages:
                subscr.Recv(msg.Channel, msg.Message)
            }
        }
        fmt.Println("EXIT")
        close(sub)
        close(quit)
        close(messages)
    }()
    return
}
```

If any attempt is made to close a channel, it instantly results in

```
panic: runtime error: send on closed channel

goroutine 12 [running]:
github.com/hoisie/redis.func·002()
.../redis/redis.go:265 +0x81
created by github.com/hoisie/redis.(*Client).sendCommands
.../redis/redis.go:271 +0x38b
```

--- a/redis.go
+++ b/redis.go
@@ -216,6 +216,7 @@ func (client *Client) sendCommands(cmdArgs <-chan []string, data chan<- interfac
     var reader *bufio.Reader
     var pong interface{}
     var errs chan error
+    var errsClosed = false
 
     if err != nil {
         goto End
@@ -251,23 +252,33 @@ func (client *Client) sendCommands(cmdArgs <-chan []string, data chan<- interfac
         for cmdArg := range cmdArgs {
             err = writeRequest(c, cmdArg[0], cmdArg[1:]...)
             if err != nil {
-                errs <- err
+                if !errsClosed {
+                    errs <- err
+                }
                 break
             }
         }
-        close(errs)
+        if !errsClosed {
+            errsClosed = true
+            close(errs)
+        }
     }()
 
     go func() {
         for {
             response, err := readResponse(reader)
             if err != nil {
-                errs <- err
+                if !errsClosed {
+                    errs <- err
+                }
                 break
             }
             data <- response
         }
-        close(errs)
+        if !errsClosed {
+            errsClosed = true
+            close(errs)
+        }
     }()
 
     // Block until errs channel closes

